src/wazuh_modules/wm_aws.c

```diff
@@ -39,40 +39,31 @@ const wm_context WM_AWS_CONTEXT = {
 void* wm_aws_main(wm_aws *aws_config) {
     wm_aws_bucket *cur_bucket;
     wm_aws_service *cur_service;
-    time_t time_start;
-    time_t time_sleep = 0;
     char *log_info;
 
 
     wm_aws_setup(aws_config);
     mtinfo(WM_AWS_LOGTAG, "Module AWS started");
 
-    // First sleeping
+    // Main loop
+
+    do {
 
-    if (!aws_config->run_on_start) {
-        time_start = time(NULL);
+        const time_t time_sleep = sched_scan_get_next_time(&(aws_config->scan_config), WM_AWS_LOGTAG, aws_config->run_on_start);
 
-        // On first run, take into account the interval of time specified
         if (aws_config->state.next_time == 0) {
-            aws_config->state.next_time = time_start + aws_config->interval;
+            aws_config->state.next_time = aws_config->scan_config.time_start + time_sleep;
```

Comment:
I think you got stuck on old code here. `scan_config.time_start` is no longer set in `sched_scan_get_next_time`. This was changed after Chema's review https://github.com/wazuh/wazuh/pull/4404#discussion_r364661199

Reply:
Yesm the `time_start` is being modified inside `sched_scan_get_next`_time but it shouldn't because of https://github.com/wazuh/wazuh/pull/4404#discussion_r364661199. I have added a second variable `last_scan_time` to avoid overriding that variable

Reply:
Good catch! `time_start` should not be overwritten between executions